docs/core-library/h3indexing.md

```diff
@@ -13,44 +13,57 @@ Child hexagons are linearly smaller than their parent hexagons.
+
+### Invalid Index
 
 Mode 0 contains a special index, `H3_NULL`, which is unique: it is bit-equivalent to `0`.
 This index indicates, *specifically*, an invalid, missing, or uninitialized **H3** index;
 it is analogous to `NaN` in floating point.
 It should be used instead of an arbitrary Mode 0 index, due to its uniqueness and easy identifiability.
 
-The components of the **H3** cell index (mode 1) are packed into a 64-bit integer in order, highest bit first, as follows:
+### H3 Cell Index
+
+An H3 Cell index (mode 1) represents a cell (hexagon or pentagon) in the H3 grid system at a particular resolution. The components of the H3 Cell index are packed into a 64-bit integer in order, highest bit first, as follows:
 
 * 1 bit reserved and set to 0,
-* 4 bits to indicate the index mode,
+* 4 bits to indicate the H3 Cell index mode,
 * 3 bits reserved and set to 0,
 * 4 bits to indicate the cell resolution 0-15,
-* 7 bits to indicate the base cell 0-121, and
+* 7 bits to indicate the base cell 0-121,
 * 3 bits to indicate each subsequent digit 0-6 from resolution 1 up to the resolution of the cell (45 bits total are reserved for resolutions 1-15)
 
-The components of the **H3** unidirectional edge index (mode 2) are packed into a 64-bit integer in order, highest bit first, as follows:
+The three bits for each unused digit are set to 7.
+
+### H3 Unidirectional Edge Index
+
+An H3 Undirectional Edge index (mode 2) represents a single directed edge between two cells (an "origin" cell and a neighboring "destination" cell). The components of the H3 Unidirectional Edge index are packed into a 64-bit integer in order, highest bit first, as follows:
 
 * 1 bit reserved and set to 0,
-* 4 bits to indicate the index mode,
-* 3 bits to indicate the edge 1-6 of the cell to traverse,
-* 4 bits to indicate the cell resolution 0-15,
-* 7 bits to indicate the base cell 0-121, and
-* 3 bits to indicate each subsequent digit 0-6 from resolution 1 up to the resolution of the cell (45 bits total are reserved for resolutions 1-15)
+* 4 bits to indicate the H3 Unidirectional Edge index mode,
+* 3 bits to indicate the edge (1-6) of the origin cell,
+* Subsequent bits matching the index bits of the origin cell.
 
-The canonical string representation of an **H3Index** is the hexadecimal representation of the integer, using lowercase letters. The string representation is variable length (no zero padding) and is not prefixed or suffixed.
+### H3 Vertex Index
 
-The three bits for each unused digit are set to 7.
+An H3 Vertex index (mode 4) represents a single topological vertex in H3 grid system, shared by three cells. Note that this does not include the distortion vertexes occasionally present in a cell's geo boundary. An H3 Vertex is arbitrarily assigned one of the three neighboring cells as its "owner". The components of the H3 Vertex index are packed into a 64-bit integer in order, highest bit first, as follows:
```

Comment:
Comment that the owner is used to calculate the canonical index of the vertex? Otherwise not clear what "owner" means.

Reply:
will do